File: API/Entities/Order/Order.cs
namespace API.Entities.Order;

public class Order
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = default!;

    public ShippingAddress ShippingAddress { get; set; } = default!;
    public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.UtcNow;
    public List<OrderItem> OrderItems { get; set; } = new();

    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }

    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
}

File: UI/web/src/app/models/order.ts
import { Address } from "./account.ts";
import { BasketItem } from "./basket.ts";

export interface ShippingAddress extends Address {}

export interface CreateOrderRequest {
  address: ShippingAddress;
  saveAddress: boolean;
}

export interface OrderItem
  extends Pick<
    BasketItem,
    "productId" | "name" | "price" | "pictureUrl" | "quantity"
  > {}

export interface Order {
  id: number;
  buyerEmail: string;
  orderDate: string;
  shippingAddress: ShippingAddress;
  deliveryFee: number;
  orderItems: OrderItem[];
  subtotal: number;
  orderStatus: string;
  total: number;
}
